markdown/design-to-code-workshop/design-to-code-workshop.md

```diff
@@ -0,0 +1,591 @@
+
+1. 🚖 Capture this high level design in the PubSub+ Event Portal where we can define our Event-Driven Architecture, including its' components: Applications, Events and Schemas. This will allow us to define the details needed to implement, visualize and extend the architecture as it evolves, and share/collaborate with our entire engineering team as we continue to innovate.  
+1. 🚕 Next up we're going to create the _RideDropoffProcessor_ microservice which will subscribe to the stream of _dropoff_ taxi updates from the fleet, capture events for a specified time window (we'll use 20 seconds to make it easy), calculate the averages, and publish a new _RideAverageUpdate_ event for each window.  
+1. 🚖 Lastly we'll create a _RideDropoffConsumer_ that receives the stream of _RideAverageUpdate_ events and captures them for display and further processing. 
+
+
+![Architecture](img/arch.webp)
+
+Positive
+: The dataset you will be using in this lab originally comes from the NYC Taxi & Limousine Commission's open data release of more than a billion taxi ride records. Google then extended one week worth of data (3M taxi rides) from their original pickup and drop-off points into full routes in order to simulate a fleet of taxis roaming the streets of NYC as they define [here](https://codelabs.developers.google.com/codelabs/cloud-dataflow-nyc-taxi-tycoon/?_ga=2.11039092.-1355519641.1572284467/#0). Solace is streaming this data over Solace PubSub+ for you to analyze and process. 
+<p>Terms of Use: This dataset is publicly available for anyone to use under the following terms provided by the Dataset Source — [https://data.cityofnewyork.us/](https://data.cityofnewyork.us/) — and is provided "AS IS" without any warranty, express or implied, from Solace. Solace disclaims all liability for any damages, direct or indirect, resulting from the use of the dataset.</p>
+
+## What You'll Need
+Duration: 0:08:00
+
+🛠 This page covers the setup needed to perform this codelab. 🛠 
+
+### AsyncAPI Generator Requirements
+✅ Install instructions available [here](https://github.com/asyncapi/generator#requirements)
+* Node.js v12.16+ 
+* npm v6.13.7+
+
+We'll install the generator itself later 👍
+
+### Spring Cloud Stream Requirements
+✅ Spring Cloud Stream just requires Java and Maven to use 🚀  
+* Java 1.8+ 
+* Maven 3.3+
+	* On mac you can `brew install maven`
+	* Other install instructions [here](https://maven.apache.org/install.html)
+* Your favorite Java IDE 💥
+
+### PubSub+ Event Broker Connection Info
+✅ The credentials below are for a public event feed that we'll use during this codelab.
+* SMF Host: `tcp://mr-d8f4yze27kt.messaging.solace.cloud:55555`
+* Message VPN: `cto-demo-virginia-azure`
+* Username: `public-taxi-user`
+* Password: `iliketaxis`
+
+✅ Note that this client-username has permissions to subscribe to `taxi/>` and `test/taxi/>` and permissinos to publish to `test/taxi/>`
```

Comment:
just thinking out loud, do we want a public account to publish to `>`? so can we have public user to explicitly publish on a predefined single (or multiple) topic(s)?